janitor/functions/select_columns.py

```diff
@@ -68,11 +68,10 @@ def select_columns(
     if level is not None:
         df_columns = df.columns
         check("level", level, [int, str])
-        full_column_list = df_columns.get_level_values(level)
-        full_column_list = _select_column_names(
-            search_column_names, df.set_axis(labels=full_column_list, axis=1)
-        )
+        df.columns = df_columns.get_level_values(level)
+        full_column_list = _select_column_names(search_column_names, df)
         full_column_list = df_columns.isin(full_column_list, level=level)
+        df.columns = df_columns
```

Comment:
@samukweku can I check that `df.columns` as assigned on line 71 is correct? It seems like you assign `df.columns` to something in Line 71 and then reassign it to something else in line 74.

Reply:
the original columns is saved as `df_columns` at the start. theassignment on line 71 reduces `df.columns` to a single column (so selection can be done). Once the matching labels are obtained, then the original columns are reassigned back in line 74, before the final dataframe is returned.

Reply:
Got it! Yes, the explanation makes sense. Before we merge, I think it's important to have the logic documented here. Would you be kind enough add in a few comments in places that you judge would help another maintainer understand the logic?

I'll mark this as the only "change requested" item. Once we're done with this one, I'm good to approve. We'll just need one other approval and then we can merge.